### ISSUE TITLE
unregister device

### ISSUE THREAD

**Issue by jschroeerluecke** [arnesson/cordova-plugin-firebase#417]
I have a settings menu where the user can set push notifications to on or off. 

What I do now is `window.FirebasePlugin.unregister();`
If I do that, than I don't get any notifications anymore, which is great. 
But if I want to register again with  `window.FirebasePlugin.getToken()` , I do not get any notifications anymore. So I can't register again.

What am I doing wrong ?

**Comment by RaphaelBadia** [comment]
Hey, I have the same issue. I only use topics and I accidently unregistered(), now I don't know how to register back. Any idea ?

**Comment by opiispanen** [comment]
Is this the same issue I found at [Stackoverflow](https://stackoverflow.com/questions/46058217/cordova-plugin-firebase-unregister-to-push-notification), because I have an issue with Android (same as in the SO post mentions). Unregistered device still gets notifications..

**Comment by jschroeerluecke** [comment]
Yes it is the same issue, I asked that question on stackoverflow.

**Comment by opiispanen** [comment]
OK! 

I just tried to fix it on Android by commenting out some code at platforms\android\src\org\apache\cordova\firebase\FirebasePlugin.java (ionic v1 project):
```javascript
private void unregister(final CallbackContext callbackContext) {
        cordova.getThreadPool().execute(new Runnable() {
            public void run() {
                try {
                    FirebaseInstanceId.getInstance().deleteInstanceId();
                    /*
                    String currentToken = FirebaseInstanceId.getInstance().getToken();
                    if (currentToken != null) {
                        FirebasePlugin.sendToken(currentToken);
                    }
                    */
                    callbackContext.success();
                } catch (Exception e) {
                    callbackContext.error(e.getMessage());
                }
            }
        });
    }
```

It seems that the code was getting a new token right after unregistering the current one. Now it's working as expected on Android at least.

**Comment by jschroeerluecke** [comment]
I've tried that with iOS by commenting out this code
`- (void)unregister:(CDVInvokedUrlCommand *)command {
    [[FIRInstanceID instanceID] deleteIDWithHandler:^void(NSError *_Nullable error){
        if (error) {
            NSLog(@"Unable to delete instance");
        } else {
            /*NSString* currentToken = [[FIRInstanceID instanceID] token];
            if (currentToken != nil) {
                [self sendToken:currentToken];
            }*/
            CDVPluginResult *pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
            [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
        }
    }];
}`

But that didn't work. I still can't register again.

**Comment by wilfredonoyola** [comment]
I have the same bug, someone has been able to solve it

**Comment by Regicollis** [comment]
I have this bug too. At least on Android, haven't tested iOS. I get a new token and continue to receive push notifications right after I've run the unregister() function.

**Comment by wilfredonoyola** [comment]
@Regicollis  I have solved it [(#522)](https://github.com/arnesson/cordova-plugin-firebase/issues/522)

**Comment by soumak77** [comment]
closing as resolved, [see comment](https://github.com/arnesson/cordova-plugin-firebase/issues/417#issuecomment-369673082)
